pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -141,50 +145,60 @@ func clusterFromProject(project string) (string, string) {
-		invalidFieldName = "userPrincipalName"
+		return field.Forbidden(fieldPath.Child("userPrincipalName"), "")
 	case oldPRTB.GroupName != newPRTB.GroupName && oldPRTB.GroupName != "":
-		invalidFieldName = "groupName"
+		return field.Forbidden(fieldPath.Child("groupName"), "")
 	case oldPRTB.GroupPrincipalName != newPRTB.GroupPrincipalName && oldPRTB.GroupPrincipalName != "":
-		invalidFieldName = "groupPrincipalName"
+		return field.Forbidden(fieldPath.Child("groupPrincipalName"), "")
 	case (newPRTB.GroupName != "" || oldPRTB.GroupPrincipalName != "") && (newPRTB.UserName != "" || oldPRTB.UserPrincipalName != ""):
-		invalidFieldName = "both user and group"
+		return field.Forbidden(fieldPath.Child("groupName", "groupPrincipalName", "userName", "userPrincipalName"), "")
+	case newPRTB.ServiceAccount != "":
+		return field.Forbidden(fieldPath.Child("serviceAccount"), "")
 	default:
 		return nil
 	}
-
-	return fmt.Errorf("cannot update %s for clusterRoleTemplateBinding %s: %w", invalidFieldName, oldPRTB.Name, admission.ErrInvalidRequest)
 }
 
 // validateCreateFields checks if all required fields are present and valid.
-func (v *Validator) validateCreateFields(newPRTB *apisv3.ProjectRoleTemplateBinding) error {
+func (v *Validator) validateCreateFields(newPRTB *apisv3.ProjectRoleTemplateBinding, fieldPath *field.Path) *field.Error {
+	if newPRTB.ServiceAccount != "" {
+		return field.Forbidden(fieldPath.Child("serviceAccount"), "")
+	}
+
 	hasUserTarget := newPRTB.UserName != "" || newPRTB.UserPrincipalName != ""
 	hasGroupTarget := newPRTB.GroupName != "" || newPRTB.GroupPrincipalName != ""
 
 	if (hasUserTarget && hasGroupTarget) || (!hasUserTarget && !hasGroupTarget) {
-		return fmt.Errorf("binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]: %w", admission.ErrInvalidRequest)
+		return field.Invalid(fieldPath.Child("groupName", "groupPrincipalName", "userName", "userPrincipalName"),
+			"combination of [userId]/[userPrincipalId] and [groupId]/[groupPrincipalId]",
+			"binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]")
 	}
 
 	if newPRTB.ProjectName == "" {
-		return fmt.Errorf("binding must have field projectName set: %w", admission.ErrInvalidRequest)
+		return field.Required(fieldPath.Child("projectName"), "")
 	}
 
 	roleTemplate, err := v.roleTemplateResolver.RoleTemplateCache().Get(newPRTB.RoleTemplateName)
 	if err != nil {
-		return fmt.Errorf("unknown reference roleTemplate '%s': %w", newPRTB.RoleTemplateName, err)
+		return field.Invalid(fieldPath.Child("roleTemplate"), roleTemplate, "")
 	}
 
 	if roleTemplate.Locked {
-		return fmt.Errorf("referenced role '%s' is locked and cannot be assigned: %w", roleTemplate.DisplayName, admission.ErrInvalidRequest)
+		return field.NotSupported(fieldPath.Child("roleTemplate"), roleTemplate.Locked, []string{"false"})
```

Comment:
I think we should use forbidden here.
> // ErrorTypeForbidden is used to report valid (as per formatting rules)
	// values which would be accepted under some conditions, but which are not
	// permitted by the current conditions (such as security policy)